Use llm_config instead of args in export_llama functions

jackzhxng opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #11162
* #11081
* #11029
* #11028

Differential Revision: [D75484927](https://our.internmc.facebook.com/intern/diff/D75484927)

jackzhxng committed:
Use llm_config instead of args in export_llama functions

[ghstack-poisoned]

jackzhxng commented:
@jackzhxng has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D75484927).

jackzhxng committed:
Update on "Use llm_config instead of args in export_llama functions"


Differential Revision: [D75484927](https://our.internmc.facebook.com/intern/diff/D75484927)

[ghstack-poisoned]

jackzhxng commented:
@jackzhxng has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D75484927).

jackzhxng committed:
Update on "Use llm_config instead of args in export_llama functions"


Differential Revision: [D75484927](https://our.internmc.facebook.com/intern/diff/D75484927)

[ghstack-poisoned]

jackzhxng commented:
@jackzhxng has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D75484927).

jackzhxng committed:
Update on "Use llm_config instead of args in export_llama functions"


Differential Revision: [D75484927](https://our.internmc.facebook.com/intern/diff/D75484927)

[ghstack-poisoned]

jackzhxng committed:
Update on "Use llm_config instead of args in export_llama functions"


Differential Revision: [D75484927](https://our.internmc.facebook.com/intern/diff/D75484927)

[ghstack-poisoned]

jackzhxng committed:
Update on "Use llm_config instead of args in export_llama functions"


Differential Revision: [D75484927](https://our.internmc.facebook.com/intern/diff/D75484927)

[ghstack-poisoned]

jackzhxng commented:
@jackzhxng has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D75484927).

jackzhxng committed:
Update on "Use llm_config instead of args in export_llama functions"


Differential Revision: [D75484927](https://our.internmc.facebook.com/intern/diff/D75484927)

[ghstack-poisoned]

jackzhxng commented:
@jackzhxng has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D75484927).

jackzhxng committed:
Update on "Use llm_config instead of args in export_llama functions"


Differential Revision: [D75484927](https://our.internmc.facebook.com/intern/diff/D75484927)

[ghstack-poisoned]

jackzhxng committed:
Update on "Use llm_config instead of args in export_llama functions"


Differential Revision: [D75484927](https://our.internmc.facebook.com/intern/diff/D75484927)

[ghstack-poisoned]

jackzhxng committed:
Update on "Use llm_config instead of args in export_llama functions"


Differential Revision: [D75484927](https://our.internmc.facebook.com/intern/diff/D75484927)

[ghstack-poisoned]

jackzhxng committed:
Update on "Use llm_config instead of args in export_llama functions"


Differential Revision: [D75484927](https://our.internmc.facebook.com/intern/diff/D75484927)

[ghstack-poisoned]

jackzhxng committed:
Update on "Use llm_config instead of args in export_llama functions"


Differential Revision: [D75484927](https://our.internmc.facebook.com/intern/diff/D75484927)

[ghstack-poisoned]

jackzhxng committed:
Update on "Use llm_config instead of args in export_llama functions"


Differential Revision: [D75484927](https://our.internmc.facebook.com/intern/diff/D75484927)

[ghstack-poisoned]

jackzhxng committed:
Update on "Use llm_config instead of args in export_llama functions"


Differential Revision: [D75484927](https://our.internmc.facebook.com/intern/diff/D75484927)

[ghstack-poisoned]

jackzhxng committed:
Update on "Use llm_config instead of args in export_llama functions"


Differential Revision: [D75484927](https://our.internmc.facebook.com/intern/diff/D75484927)

[ghstack-poisoned]

jackzhxng committed:
Update on "Use llm_config instead of args in export_llama functions"


Differential Revision: [D75484927](https://our.internmc.facebook.com/intern/diff/D75484927)

[ghstack-poisoned]

jackzhxng committed:
Update on "Use llm_config instead of args in export_llama functions"


Differential Revision: [D75484927](https://our.internmc.facebook.com/intern/diff/D75484927)

[ghstack-poisoned]

larryliu0820 commented on backends/arm/test/models/test_llama.py:
```diff
@@ -89,8 +92,9 @@ def prepare_model(self):
         ]
         parser = build_args_parser()
         args = parser.parse_args(args)
+        llm_config = convert_args_to_llm_config(args)
```
Like my comment in previous PR please do `LlmConfig.from_args(args)`

larryliu0820 commented on examples/models/llama/config/llm_config_utils.py:
```diff
@@ -26,92 +26,143 @@ def convert_args_to_llm_config(args: argparse.Namespace) -> LlmConfig:
-    llm_config.base.metadata = args.metadata
-    llm_config.base.use_lora = bool(args.use_lora)
-    llm_config.base.fairseq2 = args.fairseq2
+    if hasattr(args, "model"):
+        llm_config.base.model_class = ModelType(args.model)
+    if hasattr(args, "params"):
+        llm_config.base.params = args.params
+    if hasattr(args, "checkpoint"):
+        llm_config.base.checkpoint = args.checkpoint
+    if hasattr(args, "checkpoint_dir"):
+        llm_config.base.checkpoint_dir = args.checkpoint_dir
+    if hasattr(args, "tokenizer_path"):
+        llm_config.base.tokenizer_path = args.tokenizer_path
+    if hasattr(args, "metadata"):
+        llm_config.base.metadata = args.metadata
+    if hasattr(args, "use_lora"):
```
Can you put these in the PR when you first introduce `llm_config_utils.py`, so that we don't have to change it here.

jackzhxng committed:
Update on "Use llm_config instead of args in export_llama functions"


Differential Revision: [D75484927](https://our.internmc.facebook.com/intern/diff/D75484927)

[ghstack-poisoned]

jackzhxng committed:
Update on "Use llm_config instead of args in export_llama functions"


Differential Revision: [D75484927](https://our.internmc.facebook.com/intern/diff/D75484927)

[ghstack-poisoned]

jackzhxng committed:
Update on "Use llm_config instead of args in export_llama functions"


Differential Revision: [D75484927](https://our.internmc.facebook.com/intern/diff/D75484927)

[ghstack-poisoned]

jackzhxng committed:
Update on "Use llm_config instead of args in export_llama functions"


Differential Revision: [D75484927](https://our.internmc.facebook.com/intern/diff/D75484927)

[ghstack-poisoned]

jackzhxng committed:
Update on "Use llm_config instead of args in export_llama functions"


Differential Revision: [D75484927](https://our.internmc.facebook.com/intern/diff/D75484927)

[ghstack-poisoned]

jackzhxng committed:
Update on "Use llm_config instead of args in export_llama functions"


Differential Revision: [D75484927](https://our.internmc.facebook.com/intern/diff/D75484927)

[ghstack-poisoned]
